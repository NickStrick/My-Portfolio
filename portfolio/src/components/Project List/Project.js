import React from 'react';
import gitLogo from '../../images/GitHub-Mark-Light-64px.png'

const Project = (props) => {
    if (!(props.project.name)) {
        props.project.name = 'ERROR'
    }
    const { img, description, name, link } = props.project

    return (

        <div className="item">
            <img src={img} />
            <div className="info">
                <h2 className='title'>{name}</h2>
                <p aria-label={'project-description'}>{description}</p>
                <div className='project-links'>
                    <img src={gitLogo} target="_blank" onClick={() => window.open(link)} />
                    {props.project.deployed && <button target="_blank" onClick={() => window.open(props.project.deployed)}>Site</button>}</div>

            </div>

        </div>
    );
}


export default Project;