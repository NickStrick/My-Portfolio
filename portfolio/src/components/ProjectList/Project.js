import React from 'react';
import gitLogo from '../../images/GitHub-Mark-Light-64px.png'

const Project = (props) => {
    if (!(props.project.name)) {
        props.project.name = 'ERROR'
    }
    const { img, description, name, link } = props.project

    function projectPage() {
        window.scrollTo({ top: 0, left: 0 })
        props.history.push(`/project/${props.index}`)
    }

    return (

        <div className="item">
            <h2 className='title'>{name}</h2>
            <div className='img-overflow'>
            <img src={img} alt={`${name} smol pic`} />
            </div>
            <div className="info">

                <p aria-label={'project-description'}>{description}</p>
                <button target="_blank" onClick={projectPage}>Learn More</button>
                {/* <div className='project-links'>

                    <img src={gitLogo} target="_blank" onClick={() => window.open(link)} />
                    {props.project.deployed && <button target="_blank" onClick={() => window.open(props.project.deployed)}>Site</button>}</div> */}

            </div>

        </div >
    );
}


export default Project;