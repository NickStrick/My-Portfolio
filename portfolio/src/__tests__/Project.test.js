import React from 'react';
import renderer from 'react-test-renderer';
import Project from '../components/Project List/Project.js';
import logo from '../images/pfp.jpg';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';

describe('Project tests', () => {
    it('renders without crashing', () => {
        const input = {
            name: "Project test",
            description: 'im making a description for my test!!!!',
            img: logo
        }
        const newProject = renderer.create(
            <Project project={input} />
        )

        expect(newProject).toMatchSnapshot();

    })

    it('renders project title', () => {
        const input = {
            name: "Project test",
            description: 'im making a description for my test!!!!',
            img: logo
        }
        const component = render(<Project project={input} />)

        const item = document.querySelector('.item')
        const title = component.getByText('Project test')

        expect(title).toHaveClass('title');
        expect(item).toContainElement(title);
        expect(title).toBeInTheDocument();
    })

    it('renders project description', () => {
        const input = {
            name: "Project test",
            description: 'im making a description for my test!!!!',
            img: logo
        }
        const component = render(<Project project={input} />)

        const item = document.querySelector('.item')
        const description = component.getByLabelText('project-description')

        expect(item).toContainElement(description);
        expect(description).toBeInTheDocument();
    })

    it('renders error when not given right information', () => {
        const input = {
            name: 0,
            description: 'im making a description for my test!!!!',
            img: logo
        }
        const component = render(<Project project={input} />)

        const title = component.getByText(/error/i)

        expect(title).toBeInTheDocument();
    })
})