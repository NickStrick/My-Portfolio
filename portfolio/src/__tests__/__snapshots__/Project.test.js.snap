// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project tests renders without crashing 1`] = `
<div
  className="item"
>
  <img
    src="pfp.jpg"
  />
  <div
    className="info"
  >
    <h2
      className="title"
    >
      Project test
    </h2>
    <p
      aria-label="project-description"
    >
      im making a description for my test!!!!
    </p>
    <button>
      &gt;    Learn More
    </button>
  </div>
</div>
`;
